File: Week4Cpp/fizzbuzz.cpp
#include <iostream>

bool divisible_by(int num1, int num2) { 
    return num1 % num2 == 0;
} 

void fizzbuzz(int limit) {
    for (int i = 1; i <= limit; i++) {
        if (divisible_by(i, 3) && divisible_by(i, 5)) {
            std::cout << "FizzBuzz\n";
        } else if (divisible_by(i, 3)) {
            std::cout << "Fizz\n";
        } else if (divisible_by(i, 5)) {
            std::cout << "Buzz\n";
        } else {
            std::cout << i << "\n";
        }
    }  
} 

int main() {
    fizzbuzz(100);
}

File: Week9Rust/fizzbuzz.rs
fn main() {
  (1..=100)
      .map(|n| match (n % 3, n % 5) {
          (0, 0) => "FizzBuzz".to_string(),
          (0, _) => "Fizz".to_string(),
          (_, 0) => "Buzz".to_string(),
          _ => n.to_string(),
      })
      .for_each(|s| println!("{}", s));
}
